pkgs/development/compilers/dotnet/default.nix

```diff
@@ -48,7 +52,19 @@ makeScopeWithSplicing' {
       };
 
     in
-    {
+    lib.optionalAttrs config.allowAliases (
+      {
+        # EOL
+        sdk_2_1 = throw "Dotnet SDK 2.1 is EOL, please use 8.0 (LTS) or 9.0 (Current)";
+        sdk_2_2 = throw "Dotnet SDK 2.2 is EOL, please use 8.0 (LTS) or 9.0 (Current)";
+        sdk_3_0 = throw "Dotnet SDK 3.0 is EOL, please use 8.0 (LTS) or 9.0 (Current)";
+        sdk_3_1 = throw "Dotnet SDK 3.1 is EOL, please use 8.0 (LTS) or 9.0 (Current)";
+        sdk_5_0 = throw "Dotnet SDK 5.0 is EOL, please use 8.0 (LTS) or 9.0 (Current)";
+      }
+      // dotnet-bin
+    )
+    // lib.mapAttrs' (k: v: lib.nameValuePair "${k}-bin" v) dotnet-bin
```

Comment:
Maybe we should add some rename warnings to let people know this happened?
I don't know how to do it without making it annoying though.
You can ignore this one if you want since I also have no idea how to do it.